CMakeLists.txt

```diff
@@ -338,6 +338,11 @@ if(NOT ("${ITK_NIFTI_IO_ANALYZE_FLAVOR}" STREQUAL "ITK4Warning" OR
 endif()
 mark_as_advanced(ITK_NIFTI_IO_ANALYZE_FLAVOR)
 
+# Allow non-orthogonal rotation matrix approximation in NIFTI files
+option(ITK_NIFTI_IO_SFORM_PERMISSIVE "Allow non-orthogonal rotation matrix in NIFTI sform by default" ON)
```

Comment:
I am very opposed to this being the default compile time option.

Was OK with a runtime selection by the developer to explicitly request to ignore the fact that this behavior is not conformant with the rest of the ITK standard.

I think it is a very dangerous proposition to allow very hard to understand data assumption violations as the default.

Reply:
Why is a compile time option needed?  The runtime behavior should be backward compatible with the historical behavior, and developers who wish to change the behavior have the option to set the desired new behavior at runtime.

PS: Thanks for putting this patch forward for discussion.

Reply:
So, I thought this way we will restore compatibility with the version before the strict checking was introduced.

Reply:
I am also apposed to a compile time option.

Per has a "GlobalDefault" runtime variable would be more appropriate? With the default as Hans has described.

Reply:
so, what's the current recommendation?

Reply:
ITK_NIFTI_IO_SFORM_PERMISSIVE defaulting to OFF? And whoever doesn't like that default, can change it at both configure time and run time.

Reply:
Additional layer of complexity would be environment variable, similar to this logic: https://github.com/InsightSoftwareConsortium/ITK/blob/v5.3.0/Modules/Core/Common/src/itkMultiThreaderBase.cxx#L287-L360